workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/MemberReflectionUtil.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace Improbable.Gdk.Core
+{
+    /// <returns>
+    ///     Helper class to list members/properties with a given attribute.
+    /// </returns>
+    public class MemberReflectionUtil
+    {
+        private readonly Type attributeType;
+
+        private const BindingFlags FieldFlags = BindingFlags.Instance | BindingFlags.NonPublic |
+                                                BindingFlags.Public | BindingFlags.DeclaredOnly;
+
+        private const BindingFlags PropertyFlags =
+            FieldFlags | BindingFlags.SetProperty | BindingFlags.GetProperty;
+
+        internal MemberReflectionUtil(Type attributeType)
+        {
+            this.attributeType = attributeType;
+        }
+
+        /// <returns>
+        ///     A MemberAdapter list of properties and fields declared in given type that match at least one of given attributes.
+        /// </returns>
+        internal List<IMemberAdapter> GetMembersWithMatchingAttributes(Type targetType)
+        {
+            List<IMemberAdapter> result = new List<IMemberAdapter>();
+            ProcessMembers(result, targetType.GetProperties(PropertyFlags), CreatePropertyAdapter);
```

Comment:
This might be worth unpacking. This is currently a lot harder to read than the 10 lines or so that you'd get if you unwrap it.

Reply:
Good point